flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -1124,19 +1124,19 @@ public static void main(final String[] args) {
         final List<CustomCommandLine> customCommandLines =
                 loadCustomCommandLines(configuration, configurationDirectory);
 
+        int retCode = 31;
         try {
             final CliFrontend cli = new CliFrontend(configuration, customCommandLines);
 
             SecurityUtils.install(new SecurityConfiguration(cli.configuration));
-            int retCode =
-                    SecurityUtils.getInstalledContext().runSecured(() -> cli.parseAndRun(args));
-            System.exit(retCode);
```

Comment:
Do you mean the Oozie will handle the system exit signal and throw an exception here? After then, we will go into the `catch` code blocks and call the system exit again.

So you are suggesting to ensure that only call the `System.exit` once. Right?

Reply:
Yes.